### ISSUE TITLE
Non-literal time intervals

### ISSUE THREAD

**Issue by illright** [phip1611/tokio-js-set-interval#5]
I ran into the restriction that I cannot pass a variable's value into the `set_interval!` macro. Are there any technical limitations that forbid that or is it possible to support this use case? I'd like to have the ability to change the interval.

```rust
let period = 5000;
set_interval!(println!("hey!"), period);
```

**Comment by phip1611** [comment]
Hi @illright,
thanks for the suggestion and the interest in my library! I'll have a look into it.

**Comment by phip1611** [comment]
Published as v1.2.1 on crates.io. Feel free to open more issues if you are missing more functionality
